### PR TITLE
copy currency code to new account

### PR DESCRIPTION
**Pull request by patrick-yan-sf**
# Critical Changes

# Changes

 - When NPSP automatically creates a Household Account from a new Contact, the newly created Account's currency field is populated with the currency field of the Contact (in organizations with multi-currency enabled). Previously, this was not the case. Thank you to Matt Parlane (@mattparlane) for your contributions to this enhancement. 

# Issues Closed
- Fixes #4126 

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by patrick-yan-sf** [comment]
**lurch: add W-031981

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000008LltzQAC" target="_blank">W-032112</a>

**Comment by patrick-yan-sf** [comment]
@mattparlane, Thank you for your contribution.  We have incorporated your suggestion with changes for orgs without multi-currency enabled.  Please let us know if there are any questions.

**Comment by mattparlane** [comment]
Vunderbar! Any idea what version it will be included in? I'd like to let my people know when it's released.

**Comment by patrick-yan-sf** [comment]
@sam-knox can you review the changes section to make sure the wording is correct . Thanks

**Comment by patrick-yan-sf** [comment]
> Vunderbar! Any idea what version it will be included in? I'd like to let my people know when it's released.

@mattparlane - It is scheduled to be in the next release

**Comment by patrick-yan-sf** [comment]
**lurch: attach W-031981

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000008LlirQAC" target="_blank">W-031981</a>

File: src/classes/ACCT_IndividualAccounts_TEST.cls [hunk 1]
```diff
@@ -89,6 +89,51 @@ private class ACCT_IndividualAccounts_TEST {
...
+    @IsTest
+    private static void accountCreatedWithCorrectCurrencyCodes() {
+        //Skip the test if Advancement is installed
+        if (ADV_PackageInfo_SVC.useAdv()) return;
+
+        Boolean isMultiCurrencyEnabled = UserInfo.isMultiCurrencyOrganization();
+        String defaultCurrency = UserInfo.getDefaultCurrency();
+
+        Contact newContact = new Contact(
+            lastname = 'CurrencyTest',
```
Comment (RadSFDeveloper) [inline src/classes/ACCT_IndividualAccounts_TEST.cls]: Let's use CamelCase for fields API names, ie `LastName` and `FirstName`.

Comment (patrick-yan-sf) [reply src/classes/ACCT_IndividualAccounts_TEST.cls]: @RadSFDeveloper Thanks.  I've updated the names

File: src/classes/ACCT_IndividualAccounts_TEST.cls [hunk 2]
```diff
@@ -89,6 +89,51 @@ private class ACCT_IndividualAccounts_TEST {
...
+        Contact newContact = new Contact(
+            LastName = 'CurrencyTest',
+            FirstName = 'Test'
+        );
+        
+        String currencyField = '';
+        if (isMultiCurrencyEnabled) {
+
+             // Set currency to be something different than the default one if possible
+            List<SObject> currencies = Database.query('SELECT IsoCode FROM CurrencyType WHERE IsActive = True');
```
Comment (RadSFDeveloper) [inline src/classes/ACCT_IndividualAccounts_TEST.cls]: How about using `UTIL_UnitTestData_TEST.nonDefaultCurrencyType` for this test? See similar tests in `PMT_Payment_TEST`.

**Comment by patrick-yan-sf** [comment]
@RadSFDeveloper - I've made the requested changes.  Thanks

**Comment by patrick-yan-sf** [comment]
@rponti-sforg This is ready for QA.  Thanks

**Pull request merged**